src/gl/gl_builders.js

```diff
@@ -145,213 +145,178 @@ GLBuilders.buildPolylines = function (
+            vertex_template[scaling_index + 2] = halfWidth;
+        }
+
+        //  Add vertex to VBO
         vertex_data.addVertex(vertex_template);
     }
 
-    // Build triangles for a single line segment, extruded by the provided width
-    function buildSegment (pa, pb) {
-        var slope = Vector.normalize([(pb[1] - pa[1]) * -1, pb[0] - pa[0]]);
-
-        var pa_outer = [pa[0] + slope[0] * width/2, pa[1] + slope[1] * width/2];
-        var pa_inner = [pa[0] - slope[0] * width/2, pa[1] - slope[1] * width/2];
+    for (var ln = 0; ln < num_lines; ln++) {
+        var line = lines[ln];
+        var lineSize = line.length;
 
-        var pb_outer = [pb[0] + slope[0] * width/2, pb[1] + slope[1] * width/2];
-        var pb_inner = [pb[0] - slope[0] * width/2, pb[1] - slope[1] * width/2];
+        if (lineSize < 2) {
+            continue;
+        }
 
-        vertices.push(
-            pb_inner, pb_outer, pa_inner,
-            pa_inner, pb_outer, pa_outer
-        );
+        //  Initialize variables
+        var normPrevCurr = [0.0,0.0,0.0],  // Right normal to segment between previous and current m_points
+            normCurrNext = [0.0,0.0,0.0],  // Right normal to segment between current and next m_points
+            rightNorm = [0.0,0.0,0.0];     // Right "normal" at current point, scaled for miter joint
+
+        var prevCoord = [], // Previous point coordinates
+            currCoord = [], // Current point coordinates
+            nextCoord = []; // Next point coordinates
+
+        //  Compute first pair of vertexes based on the second point of the line
+        currCoord = Vector.set( line[0] );
+        nextCoord = Vector.set( line[1] );
+
+        //  a) Get perpendicular of the vector between first and second point
+        normCurrNext = Vector.normalize( Vector.perp( currCoord, nextCoord ) );
+        rightNorm = normCurrNext;
+
+        //  b) add the pair vertex into the vertex buffer
+        if (scaling_index) {
+            //  b.a If scaling is on add the vertex (the currCoord) and the scaling Vecs (normals pointing where to extrude the vertexes)
+            vertices.push(currCoord);
+            vertices.push(currCoord);
+            scalingVecs.push( rightNorm );
+            scalingVecs.push( Vector.neg(rightNorm) );
+        } else {
+            //  b.b Add the extruded vertexes
+            vertices.push([ currCoord[0] + rightNorm[0] * halfWidth,
+                            currCoord[1] + rightNorm[1] * halfWidth ]);
+            vertices.push([ currCoord[0] - rightNorm[0] * halfWidth,
+                            currCoord[1] - rightNorm[1] * halfWidth ]);
+        }
 
-        // Add UVs
+        // c) Add uv's if they are enable
         if (texcoord_index) {
-            texcoords.push(
-                [min_u, min_v], [max_u, min_v], [min_u, max_v],
-                [min_u, max_v], [max_u, min_v], [max_u, max_v]
-            );
+            texcoords.push( [max_u, min_v], [min_u, min_v] );
```

Comment:
Is this right? 3 references to min but only 1 to max? (I might just not see what's happening here.)